Sources/ClientRuntime/Util/Async.swift

```diff
@@ -0,0 +1,53 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Dispatch
+
+private class BoxedThrowingResult<ResultType> {
+    var result: Result<ResultType, Error>?
+}
+
+private class BoxedResult<ResultType> {
+    var result: ResultType?
+}
+
+/// Unsafely awaits a throwing async function from a synchronous context.
+@available(*, deprecated, message: """
+    This is a temporary workaround for the lack of async support.
+    It will be migrated to structured concurrency in the future
```

Comment:
When "in the future" do you think this will be migrated to structured concurrency? What's preventing us from doing it now?

Reply:
underlying CRT APIs. I'm going to create issues for them.

Sometimes protocol properties like `CustomDebugStringConvertible` which can't be made async implementation.